Adding connectivity test to troubleshooting doc

jedsalazar opened the pull request:
Previously the troubleshooting documentation lacked a reference to the connectivity tests specified in the getting started guide.

Added the connectivity guide reference as well as some minor style updates

Fixes: #11581

Signed-off-by: Jed Salazar <jed@isovalent.com>

Please ensure your pull request adheres to the following guidelines:

- [x] For first time contributors, read [Submitting a pull request](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#submitting-a-pull-request)
- [ ] All code is covered by unit and/or runtime tests where feasible.
- [ ] All commits contain a well written commit description including a title,
      description and a `Fixes: #XXX` line if the commit addresses a particular
      GitHub issue.
- [x] All commits are signed off. See the section [Developer’s Certificate of Origin](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#dev-coo)
- [ ] Provide a title or release-note blurb suitable for the release notes.
- [ ] Thanks for contributing!

<!-- Description of change -->

Fixes: #issue-number

```release-note
Add connectivity test to troubleshooting
```

Commit:
Adding connectivity test to troubleshooting doc

Previously the troubleshooting documentation lacked a reference to the connectivity tests specified in the getting started guide.

Added the connectivity guide reference as well as some minor style updates

Fixes: #11581

Signed-off-by: Jed Salazar <jed@isovalent.com>

Updating issues in the docs

Fixing ts_agent_logs line, adding a link to the test and fixing a spelling error

joestringer commented on Documentation/troubleshooting.rst:
```diff
@@ -137,15 +142,53 @@ e.g.:
+Cilium connectivity tests
+------------------------------------
+
+The Cilium connectivity test_ deploys a series of services, deployments, and 
```
Two comments:

* Did you intend to explicitly describe to the user how to deploy these? That's at least how the other docs reference the connectivity-check YAML. Note that there is already a template of that text in `Documentation/gettingstarted/k8s-install-connectivity-test.rst`. We should be able to just import that docs fragment here.

* Secondly, this will only work in a namespace by itself with no other pods or network policies applied. If there is a Cilium Clusterwide Network Policy enabled, that may also break this connectivity check.

jedsalazar replied:
Thanks, Joe. 
> * Did you intend to explicitly describe to the user how to deploy these? That's at least how the other docs reference the connectivity-check YAML. Note that there is already a template of that text in `Documentation/gettingstarted/k8s-install-connectivity-test.rst`. We should be able to just import that docs fragment here.

This section is a slightly modified copy of that docs fragment. I updated only to describe what gets deployed but the rest of the text is verbatim from that template. 

> * Secondly, this will only work in a namespace by itself with no other pods or network policies applied. If there is a Cilium Clusterwide Network Policy enabled, that may also break this connectivity check.

That's a great point. Given that the test deploys to the default namespace, this could cause problems in a production environment. We could either communicate that caveat or change the namespace that it's deployed to (something unlikely to be in use like `cilium-test`?). How is this generally handled in the wild today? Do we add the `Namespace` stanza or avoid suggesting these tests in production environments?

joestringer replied:
> This section is a slightly modified copy of that docs fragment. I updated only to describe what gets deployed but the rest of the text is verbatim from that template.

Are we able to share more here? I'm mainly concerned about when we inevitably make changes to these common fragments, and have to go and update multiple places. Whereas if we're able to tie it down to the one fragment file, we can just fix up the one example.

> We could either communicate that caveat or change the namespace that it's deployed to (something unlikely to be in use like cilium-test?). How is this generally handled in the wild today? Do we add the Namespace stanza or avoid suggesting these tests in production environments?

I like the `cilium-test` namespace idea. We may consider actually just changing the namespace in the YAMLs themselves to achieve this.

As for today, I think we just have the instructions in the "getting started guides" and infer that users won't run them in a sensitive namespace.

jedsalazar replied:
> > This section is a slightly modified copy of that docs fragment. I updated only to describe what gets deployed but the rest of the text is verbatim from that template.
> 
> Are we able to share more here? I'm mainly concerned about when we inevitably make changes to these common fragments, and have to go and update multiple places. Whereas if we're able to tie it down to the one fragment file, we can just fix up the one example.
> 

I like this idea. I'm going to develop a breakdown of each test and specify what type of traffic it's validating. I'll update the testing definitions here and we can decide where one fragment will live and can be referenced. Let me know if you think I should move it to a dedicated location. 

> > We could either communicate that caveat or change the namespace that it's deployed to (something unlikely to be in use like cilium-test?). How is this generally handled in the wild today? Do we add the Namespace stanza or avoid suggesting these tests in production environments?
> 
> I like the `cilium-test` namespace idea. We may consider actually just changing the namespace in the YAMLs themselves to achieve this.
> 
> As for today, I think we just have the instructions in the "getting started guides" and infer that users won't run them in a sensitive namespace.

That inference makes sense for a getting started guide, but once this is published to a troubleshooting guide, we should be more clear on the impacts the testing could have on their system. We can give a high-level caveat about cluster-wide network policies, but I'm curious about how you think we should handle the namespacing. Do you think we should adjust the namespace this is deployed to directly in the yaml or for the purposes of this doc we can have the operator create the cilium-test namespace and deploy with `k apply --namespace=cilium-test -f https://..connectivity-check.yaml`?

joestringer reviewed (changes requested):
This is a welcome change :)

Couple of comments below.

michi-covalent commented on Documentation/troubleshooting.rst:
```diff
@@ -137,15 +142,53 @@ e.g.:
+variant and the readiness and liveness gate indicates success or failure of the
+test:
+
+.. _test: https://raw.githubusercontent.com/cilium/cilium/1.7.4/examples/kubernetes/connectivity-check/connectivity-check.yaml
```
i think you need to use the `SCM_WEB` variable defined in Documentation/conf.py here to avoid hardcoding branches/tags. here is an example: https://github.com/cilium/cilium/blame/054dd163fdb792c44b3c493b8d6fd111e5e1a2bc/Documentation/gettingstarted/dns.rst#L33

michi-covalent reviewed (commented):
lgtm overall, one minor comment about not hardcoding the release tag.

Commit:
Update link to reference SCM_WEB var

Hard-coding is bad, variables are good.

Fixes: #11581

Signed-off-by: Jed Salazar <jed@isovalent.com>

jedsalazar reviewed (commented):
> lgtm overall, one minor comment about not hardcoding the release tag.

thanks, Michi! added. it doesn't seem to render on my local build. going see if it works in netlify

Commit:
Adding cilium-test namespace creation and depth of test descriptions

Given that the connectivity testing is referenced in the troubleshooting docs, it's worth deploying them to an empty namespace, `cilium-test`. h/t @Joe

Also included an initial effort to inlcude a table describing the coverage that each testing type provides.

Fixes: #11581

Signed-off-by: Jed Salazar <jed@isovalent.com>

Commit:
Fix spelling error and add intra to wordlist

Fixes: #11581

Signed-off-by: Jed Salazar <jed@isovalent.com>

michi-covalent reviewed (approved):
let's roll
